frontend/public/components/terminal.jsx

```diff
@@ -3,15 +3,18 @@ import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import { Terminal as XTerminal } from 'xterm';
 import * as fit from 'xterm/lib/addons/fit/fit';
+import * as fullscreen from 'xterm/lib/addons/fullscreen/fullscreen';
 
 XTerminal.applyAddon(fit);
+XTerminal.applyAddon(fullscreen);
 
 export class Terminal extends React.Component {
   constructor(props) {
     super(props);
     this.state = {height: 0, width: 0};
     this.innerRef = React.createRef();
     this.outerRef = React.createRef();
+    this.inFullScreenMode = false;
```

Comment:
I suggest `isFullscreen`. It's shorter and follows the convention of booleans in the format `isFoo`

Fullscreen is one word so I'd prefer not to capitalize the `s`. I know the xterm uses camel case there, but HTML5 fullscreen APIs don't, which we'll probably use elsewhere at some point (e.g., for logs). Either way we'll probably have some inconsistency.